AnaProd/anaTupleProducer.py

```diff
@@ -115,12 +119,24 @@ def addAllVariables(dfw, syst_name, isData, trigger_class):
                                     f"tau{leg_idx+1}_recoJetMatchIdx>=0 ? static_cast<float>(Jet_p4.at(tau{leg_idx+1}_recoJetMatchIdx).Phi()) : -1.f;")
         dfw.DefineAndAppend( f"tau{leg_idx+1}_seedingJet_mass",
                                     f"tau{leg_idx+1}_recoJetMatchIdx>=0 ? static_cast<float>(Jet_p4.at(tau{leg_idx+1}_recoJetMatchIdx).M()) : -1.f;")
+        dfw.DefineAndAppend( f"tau{leg_idx+1}_seedingJet_e",
+                                    f"tau{leg_idx+1}_recoJetMatchIdx>=0 ? static_cast<float>(Jet_p4.at(tau{leg_idx+1}_recoJetMatchIdx).E()) : -1.f;")
 
 
+        dfw.DefineAndAppend(f"b{leg_idx+1}_idx", f"HbbCandidate.leg_index[{leg_idx}]")
         dfw.DefineAndAppend(f"b{leg_idx+1}_pt", f"static_cast<float>(HbbCandidate.leg_p4[{leg_idx}].Pt())")
+        dfw.DefineAndAppend(f"b{leg_idx+1}_pt_raw", f"static_cast<float>(Jet_pt.at(HbbCandidate.leg_index[{leg_idx}]))")
         dfw.DefineAndAppend(f"b{leg_idx+1}_eta", f"static_cast<float>(HbbCandidate.leg_p4[{leg_idx}].Eta())")
         dfw.DefineAndAppend(f"b{leg_idx+1}_phi", f"static_cast<float>(HbbCandidate.leg_p4[{leg_idx}].Phi())")
         dfw.DefineAndAppend(f"b{leg_idx+1}_mass", f"static_cast<float>(HbbCandidate.leg_p4[{leg_idx}].M())")
+        dfw.DefineAndAppend(f"b{leg_idx+1}_e", f"static_cast<float>(HbbCandidate.leg_p4[{leg_idx}].E())")
+        if not isData:
+            dfw.DefineAndAppend(f"b{leg_idx+1}_genJet_idx", f"Jet_genJetIdx.at(HbbCandidate.leg_index[{leg_idx}])")
+            dfw.DefineAndAppend(f"b{leg_idx+1}_genJet_pt", f"GenJet_p4.at(b{leg_idx+1}_genJet_idx).Pt()")
+            dfw.DefineAndAppend(f"b{leg_idx+1}_genJet_eta", f"GenJet_p4.at(b{leg_idx+1}_genJet_idx).Eta()")
+            dfw.DefineAndAppend(f"b{leg_idx+1}_genJet_phi", f"GenJet_p4.at(b{leg_idx+1}_genJet_idx).Phi()")
+            dfw.DefineAndAppend(f"b{leg_idx+1}_genJet_mass", f"GenJet_p4.at(b{leg_idx+1}_genJet_idx).M()")
+            dfw.DefineAndAppend(f"b{leg_idx+1}_genJet_e", f"GenJet_p4.at(b{leg_idx+1}_genJet_idx).E()")
```

Comment:
- don't store E
- check that index >= 0

Reply:
ok